Cargo.toml

```diff
@@ -87,7 +87,7 @@ libp2p-tcp = { version = "0.19.0", path = "transports/tcp", optional = true }
 libp2p-websocket = { version = "0.19.0", path = "transports/websocket", optional = true }
 
 [dev-dependencies]
-async-std = "1.0"
+async-std = "< 1.6"
```

Comment:
Would this allow versions below `1.0` to be used? If so how about `>=1.0, < 1.6`.

Reply:
I can't find anything about that in the docs, but it might indeed be the case

Reply:
I don't think it really matters as cargo prefers newer versions, doesn't it? Do you think the lower bound is important?

Reply:
There's a general problem in the Rust ecosystem: you might be using features introduced in `foo v1.1` but only have `foo v1.0` in your dependencies. Cargo is theoretically allowed to depend on `foo v1.0`, which will make the compilation fail.

This issue is a bit everywhere, but people don't realize it because as you said Cargo prefers newer versions.

I'd say we're in the same situation here, so I'm tempted to not care.

Reply:
Mmh, due to semver libp2p selects a version smaller than 1.0, so 0.99 gets selected currently. I think the correct fix would be `=1.5`

Reply:
Yeah we've actually been bitten by it in Polkadot and released two versions with an old async-std (that had the bug with a high number of threads being spawned) without realizing.